test/inductor/test_cpu_repro.py

```diff
@@ -2765,6 +2771,23 @@ def forward(self, x):
                 "Vectorized<float>::loadu(tmpbuf.data())", 0, exactly=True
             ).run(code)
 
+    @parametrize("dtype", (torch.float16, torch.bfloat16, torch.float))
+    @parametrize("shape", ("15,3,13", "4,2048,4096"))
+    def test_fp8_cast(self, dtype: torch.dtype, shape: str):
+        def fp8_cast(x):
+            y0 = x.to(dtype=torch.float8_e4m3fn).to(dtype)
+            y1 = x.to(dtype=torch.float8_e5m2).to(dtype)
+            return y0, y1
+
+        compiled_fp8_cast = torch.compile(fp8_cast, backend="inductor", dynamic=True)
+
+        shape = [int(dim) for dim in shape.split(",")]
+        x = torch.rand(*shape, device="cpu", dtype=dtype)
+        y0_fp8, y1_fp8 = compiled_fp8_cast(x)
+
+        torch.testing.assert_close(y0_fp8, x, rtol=5e-1, atol=5e-1)
+        torch.testing.assert_close(y1_fp8, x, rtol=5e-1, atol=5e-1)
```

Comment:
why not just comparing with eager like other tests do?

Reply:
I am copying it from the `CUDA` test before. Now I aligned it with other tests now (using `self.common`).